lib/mongoose_push/service/apns/supervisor.ex

```diff
@@ -72,4 +98,69 @@ defmodule MongoosePush.Service.APNS.Supervisor do
 
     [single_config | mode_list]
   end
+
+  defp convert_token_pool_to_sparrow({pool_name, pool_config}, {mode_list, tokens}) do
+    auth_type = {:auth_type, :token_based}
+    pool_size = {:worker_num, pool_config[:pool_size]}
+
+    endpoint_mode = @default_endpoints[pool_config[:mode]]
+    endpoint = {:endpoint, pool_config[:endpoint] || endpoint_mode}
+
+    port_config =
+      case pool_config[:use_2197] do
+        true ->
+          2197
+
+        _ ->
+          nil
+      end
+
+    port = {:port, port_config}
+
+    name = {:pool_name, pool_name}
+
+    key = pool_config[:auth][:key_id]
+    team = pool_config[:auth][:team_id]
+    p8_file_path = pool_config[:auth][:p8_file_path]
+
+    if not File.exists?(p8_file_path) do
+      Logger.error(~s"Required file does not exist in the given path:
+      p8_file=#{p8_file_path}")
+
+      Supervisor.stop(self(), {:error, "Required file does not exist in the given path:
+      p8_file=#{p8_file_path}"})
+    end
+
+    if is_nil(key) or is_nil(team) do
```

Comment:
Why not doing the full check for all config opts?
Checkout PoC: https://github.com/esl/sparrow/pull/4
I would highly recommend to add check config responsibility to be moved to https://github.com/esl/sparrow and do it on starting sparrow.